openeo/rest/datacube.py

```diff
@@ -187,9 +193,17 @@ def load_collection(
                     "Unexpected parameterized `spatial_extent` in `load_collection`:"
                     f" expected schema with type 'object' but got {spatial_extent.schema!r}."
                 )
+        elif not spatial_extent or (isinstance(spatial_extent, dict) and spatial_extent.keys() & {"west", "east", "north", "south"}):
```

Comment:
to make it a bit less confusing:
```suggestion
        elif spatial_extent is None or (isinstance(spatial_extent, dict) and spatial_extent.keys() & {"west", "east", "north", "south"}):
```